APE26.rst

```diff
@@ -0,0 +1,220 @@
+This duplicates functionality with ``SkyCoord``, which acts as a container for both 
+coordinate data and reference frame information. We propose to change the frame classes 
+such that they only store metadata and never coordinate data. This would make the 
+implementation more modular, remove ambiguity for users from having nearly duplicate 
+functionality with slightly different APIs, and better satisfy the principle of 
+Separation of Concerns.
+
+Detailed description
+--------------------
+The coordinate frame classes (subclasses of ``BaseCoordinateFrame``, e.g., ``ICRS``, 
+``Galactic``, ``FK4``, etc.) are used to represent astronomical reference frames. These 
+may also contain position, velocity, time, or other information about the reference frame 
+relative to another reference frame, which is used to transform coordinates between the 
+reference frames that exist in the ``astropy.coordinates`` ecosystem. For example, the 
+``AltAz`` frame class, which is used when referencing sky coordinates in altitude and 
+azimuth, must contain a location on Earth and a time in order to transform to and from a 
+fixed reference frame like the ICRS. In addition to reference frame information, these 
+frame classes can **also** contain coordinate data (positions, velocities, or other 
+differentials), which are stored internally using the ``Representation`` and 
+``Differential`` classes.
+
+The above reveals a problematic oddity with the coordinate frame implementation – an 
+insufficient `separation of concerns <https://en.wikipedia.org/wiki/Separation_of_concerns>`_. 
+The coordinate frame class deals with two separate issues: the definition of the frame 
+and the storage of coordinate data within that frame. Ideally the code would be more 
+modularly structured, where the coordinate frame class **only** defines the reference 
+frame. It would then be the purview of another class – `SkyCoord` – to bring these 
+concerns together and to represent data (using `Representation` and `Differential` 
+classes) in a given reference frame (using a `CoordinateFrame` class). As a demonstration 
+of the current state of duplicated functionality, these two initializations effectively 
+represent the same thing, but return different objects with similar but not identical 
+APIs:
+
+.. code-block:: python
+
+    c1 = SkyCoord(1., 2., frame="icrs", units="deg")
+    c2 = ICRS(1. * u.deg, 2. * u.deg)
+
+We suggest the situation should instead be analogous to what is the case for units, 
+which know how to transform from one to another, but are not concerned with how the 
+values are stored (that belongs to ``Quantity``). Translating to coordinates, units are 
+like ``CoordinateFrame``, the values are the coordinate data (``Representations``), and 
+``Quantity`` is like ``SkyCoord``.
+
+Having both the frame classes as well as ``SkyCoord`` be able to store and handle data 
+has resulted in a large amount of code duplication. Restructuring the ``Frame`` classes 
```

Comment:
Not just code. It's even worse in tests because we have to test both `SkyCoord` and `BaseCoordinateFrame` methods with both `SkyCoord` and `BaseCoordinateFrame` inputs. Not testing all the combinations has led to bugs going unnoticed, e.g. https://github.com/astropy/astropy/pull/15659